File: PieGallery/app/src/main/java/com/deadely/piegallery/database/dao/UserDao.java
package com.deadely.piegallery.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.deadely.piegallery.dataclasses.User;

import java.util.List;

@Dao
public interface UserDao {

    @Query("SELECT * FROM user_table")
    List<User> getAllUsers();

    @Query("SELECT * FROM user_table WHERE user_id  LIKE :id")
    User getById(long id);

    @Query("DELETE FROM user_table")
    public void deleteAllUsers();

    @Insert
    void insert(User user);

    @Update
    void update(User user);

    @Delete
    void delete(User user);

}

File: itlen/app/src/main/java/com/deadely/itl_en/database/dao/UserDao.kt
package com.deadely.itl_en.database.dao

import androidx.room.*
import com.deadely.itl_en.dataclasses.User

@Dao
interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addUser(user: User)

    @Update
    fun updateUser(user: User)

    @Delete
    fun deleteUser(user: User)

    @Query("DELETE FROM user_table")
    fun deleteAllUsers()

    @Query("SELECT * FROM user_table WHERE user_id LIKE :id LIMIT 1")
    fun getUserById(id: String): User

    @Query("SELECT * FROM user_table WHERE active LIKE :active LIMIT 1")
    fun getActiveUser(active: Boolean): User


    @Query("SELECT * FROM user_table")
    fun getAllUsers(): List<User>

    @Query("SELECT * FROM user_table WHERE email LIKE :email AND password LIKE :pass")
    fun getByEmailAndPass(email: String, pass: String): User


}
